liquibase-standard/src/main/java/liquibase/util/ObjectMethods.java

```diff
@@ -0,0 +1,45 @@
+package liquibase.util;
+
+import static java.util.Locale.ROOT;
+import static java.util.stream.Collectors.toMap;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BinaryOperator;
+
+final class ObjectMethods {
+
+  private static final BinaryOperator<Method> ignoreMultipleMethodsWithSameName = (first, second) -> first;
+  private final Map<String, Method> writeMethods;
+  private final Map<String, Method> readMethods;
+
+  ObjectMethods(Method[] methods) {
+    readMethods = new HashMap<>(methods.length);
+    readMethods.putAll(find(methods, 0, "get"));
+    readMethods.putAll(find(methods, 0, "is"));
+    writeMethods = find(methods, 1, "set");
+  }
+
+  private Map<String, Method> find(Method[] methods, int parametersCount, String prefix) {
+    return Arrays.stream(methods)
+      .filter(m -> m.getParameterTypes().length == parametersCount)
+      .filter(m -> m.getName().startsWith(prefix))
+      .collect(toMap(m -> propertyName(m.getName(), prefix), m -> m, ignoreMultipleMethodsWithSameName));
+  }
+
+  private String propertyName(String methodName, String prefix) {
+    int length = prefix.length();
+    return methodName.substring(length, length + 1).toLowerCase(ROOT) + methodName.substring(length + 1);
```

Comment:
I'm not very familiar with this piece of the code. Still, I was wondering whether we should be uppercasing the given string as it was before (`.toUpperCase(Locale.ENGLISH)`) or should lowercase and this implementation does (`.toLowerCase(ROOT)`). I'm not sure if it could have any side effects, I'll check with the team.

Reply:
Hey Andrei, yeap but what I meant is previously the below code:
```
String getMethodName = "get" + propertyName.substring(0, 1).toUpperCase(Locale.ENGLISH)
            + propertyName.substring(1);
```
would return `NAME` for a get/is method name.

Reply:
@MalloD12 No, it returns `getName`.
Method `toUpperCase` is called only for the first letter after "get".

Reply:
Hi @asolntsev, 

My bad you are right. Is there a reason or just an implementation agreement you made for returning `name`, instead of `Name`?

Reply:
> Is there a reason or just an implementation agreement you made for returning name, instead of Name?

@MalloD12 Sorry, I don't understand your last question. Given `methodName = "getName"`, method `propertyName("getName", "get")` returns string `name`. It's the property.

Reply:
I was asking if there was a reason why we are returning `name` and not `Name`.